src/CoreWCF.Primitives/src/CoreWCF/Channels/LocalServiceSecuritySettings.cs

```diff
@@ -0,0 +1,377 @@
+                        SR.Format(SR.SFxTimeoutOutOfRangeTooBig)));
+                }
+
+                this.sessionKeyRolloverInterval = value;
+            }
+        }
+
+        public bool ReconnectTransportOnFailure
+        {
+            get
+            {
+                return this.reconnectTransportOnFailure;
+            }
+            set
+            {
+                this.reconnectTransportOnFailure = value;
+            }
+        }
+
+        public TimeSpan TimestampValidityDuration
+        {
+            get
+            {
+                return this.timestampValidityDuration;
+            }
+            set
+            {
+                if (value < TimeSpan.Zero)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value", value,
+                        SR.Format(SR.SFxTimeoutOutOfRange0)));
+                }
+
+                if (TimeoutHelper.IsTooLarge(value))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value", value,
+                        SR.Format(SR.SFxTimeoutOutOfRangeTooBig)));
+                }
+
+                this.timestampValidityDuration = value;
+            }
+        }
+
+        public int MaxCachedCookies
+        {
+            get
+            {
+                return this.maxCachedCookies;
+            }
+            set
+            {
+                if (value < 0)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("value", value,
+                                                    SR.Format(SR.ValueMustBeNonNegative)));
+                }
+                this.maxCachedCookies = value;
+            }
+        }
+
+        public LocalServiceSecuritySettings()
+        {
+            this.DetectReplays = SecurityProtocolFactory.defaultDetectReplays;
+            this.ReplayCacheSize = SecurityProtocolFactory.defaultMaxCachedNonces;
+            this.ReplayWindow = SecurityProtocolFactory.defaultReplayWindow;
+            this.MaxClockSkew = SecurityProtocolFactory.defaultMaxClockSkew;
+            this.NegotiationTimeout = TimeSpan.Parse(defaultServerMaxNegotiationLifetimeString, CultureInfo.InvariantCulture);
+           this.IssuedCookieLifetime  = TimeSpan.Parse(defaultServerIssuedTokenLifetimeString, CultureInfo.InvariantCulture);
+             
+            //this.IssuedCookieLifetime = NegotiationTokenAuthenticator<NegotiationTokenAuthenticatorState>.defaultServerIssuedTokenLifetime;
+            //this.MaxStatefulNegotiations = NegotiationTokenAuthenticator<NegotiationTokenAuthenticatorState>.defaultServerMaxActiveNegotiations;
+            //this.NegotiationTimeout = NegotiationTokenAuthenticator<NegotiationTokenAuthenticatorState>.defaultServerMaxNegotiationLifetime;
```

Comment:
Why isn't MaxStatefulNegotiations needed? I found more than one place it's used that I would to get used for CoreWCF. One place is ServceCredntialsSecurityTokenManager.CreateSecureConversationTokenAuthenticator where it's used when creating an instance of SecurityContextSecurityTokenResolver